models/issue.go

```diff
@@ -1498,3 +1512,41 @@ func updateIssue(e Engine, issue *Issue) error {
 func UpdateIssue(issue *Issue) error {
 	return updateIssue(x, issue)
 }
+
+// Get Blocked By Dependencies, aka all issues this issue is blocked by.
+func (issue *Issue) getBlockedByDependencies(e Engine) (issueDeps []*Issue, err error) {
+	if err = e.
+		Table("issue_dependency").
+		Select("issue.*").
+		Join("INNER", "issue", "issue.id = issue_dependency.dependency_id").
+		Where("issue_id = ?", issue.ID).
+		Find(&issueDeps); err != nil {
+		return issueDeps, err
+	}
+
+	return
+}
+
+// Get Blocking Dependencies, aka all issues this issue blocks.
+func (issue *Issue) getBlockingDependencies(e Engine) (issueDeps []*Issue, err error) {
+	if err := e.
+		Table("issue_dependency").
+		Select("issue.*").
+		Join("INNER", "issue", "issue.id = issue_dependency.issue_id").
+		Where("dependency_id = ?", issue.ID).
+		Find(&issueDeps); err != nil {
+		return issueDeps, err
```

Comment:
I think `return issueDeps, err` and just `return` is same here, you do not need any if. When using `(issueDeps []*Issue, err error)` code is hard to read and error prone (What if you have some non important/recoverable error inside function that you want to skip? Do you want to return every variable defined all the time?).

Reply:
Seems logic, I removed the if and replaced it with a single `return`